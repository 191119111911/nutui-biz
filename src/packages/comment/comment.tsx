import React, { FunctionComponent, HTMLAttributes, ReactNode } from "react";
import { IComponent } from "@/utils/typings";
import { CommentHeader } from "./components/CommentHeader";
import { CommentBottom } from "./components/CommentBottom";
import { CommentImages } from "./components/CommentImages";
import { VideosType, ImagesType, GoodsClickParams } from "./components/CommentImages";
import bem from '@/utils/bem'
import { Icon } from "@nutui/nutui-react";

export interface CommentInfo {
  content: string, // 评论详情
  nickName: string, // 评论人的姓名
  score: number, // 评论星星数
  avatar: string, // 评论人头像
  time: string, // 评论时间
  size?: string, // 评论人购买的商品规格
  reply?: number, // 此评论的回复数
  like?: number, // 此评论的点赞数
}
export interface CommentFollow {
  days: number, // 购买多少天后进行追评
  content: string, // 追评内容
  images?: string[], // 追评图片
}
export interface CommentProps extends IComponent {
  type: "default" | "complex";
  imagesRows: "one" | "multi";
  ellipsis: string | number;
  videos: Array<VideosType>;
  images: Array<ImagesType>;
  info: CommentInfo;
  follow: CommentFollow;
  operation: Array<string>;
  commentLabels: ReactNode;
  commentShopReply: ReactNode;
  onClickOperate: (type: string) => void;
  onClick: (info: any) => void;
  onClickImages: (imgs: GoodsClickParams) => void;
}

const defaultProps = {
  type: "default",
  imagesRows: "one",
  ellipsis: 2,
  videos: new Array(),
} as CommentProps;

export const Comment: FunctionComponent<
  Partial<CommentProps> & HTMLAttributes<HTMLDivElement>
> = (props) => {
  const {
    type,
    imagesRows,
    ellipsis,
    videos,
    images,
    info,
    follow,
    operation,
    commentLabels,
    commentShopReply,
    onClickOperate,
    onClick,
    onClickImages,
  } = {
    ...defaultProps,
    ...props,
  };

  const b = bem('comment')

  const onHandleClick = () => {
    onClick && onClick(info);
  };

  const conEllipsis = () => {
    if (ellipsis) return ellipsis;
    return type === "complex" ? 6 : 2;
  };

  return (
    <div className={b()}>
      <CommentHeader
        type={type}
        info={info}
        onHandleClick={onHandleClick}
        commentLabels={commentLabels}
      ></CommentHeader>

      {info && info.content && (
        <div
          className={b('main')}
          style={{
            WebkitLineClamp: conEllipsis(),
          }}
          onClick={onHandleClick}
        >
          {" "}
          {info.content}{" "}
        </div>
      )}

      <CommentImages
        images={images}
        videos={videos}
        type={imagesRows}
        onClickImages={onClickImages}
      ></CommentImages>

      {follow && follow.days > 0 && (
        <div className={b("follow")} onClick={onHandleClick}>
          <div className={b("follow-title")}>
            购买 {follow.days} 天后追评
          </div>
          <div className={b("follow-com")}>{follow.content}</div>
          {follow.images && follow.images.length > 0 && (
            <div className={b("follow-img")}>
              {follow.images.length} 张追评图片
              <Icon name="right" width="12px"></Icon>
            </div>
          )}
        </div>
      )}

      <CommentBottom
        type={type}
        info={info}
        operation={operation}
        onClickOperate={onClickOperate}
        onHandleClick={onHandleClick}
      ></CommentBottom>

      {commentShopReply}
    </div>
  );
};

Comment.defaultProps = defaultProps;
Comment.displayName = "NutComment";
