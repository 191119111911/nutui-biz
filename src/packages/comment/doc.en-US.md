# Comment

### Intro

Used to display the comment list

### Install

```javascript
import { Comment } from '@nutui/nutui-biz';
```

## Demo

### Single Line Image

By default, images of reviews for individual items are displayed in a single slide

:::demo
```js
import React, { useState } from 'react';
import { Comment } from '@nutui/nutui-biz';

const App = () => {

  const [cmt, setCmt] = useState({} as any);

  const getData = () => {
    fetch("//storage.360buyimg.com/nutui/3x/comment_data.json")
      .then((response) => response.json())
      .then((res) => {
        res.Comment.info.avatar =
          "https://img14.360buyimg.com/imagetools/jfs/t1/167902/2/8762/791358/603742d7E9b4275e3/e09d8f9a8bf4c0ef.png";
        setCmt(res.Comment);
      })
      .catch((err) => console.log("Oh, error", err));
  };

  useEffect(() => {
    getData();
  }, []);

  const onClickImages = (imgs: any) => {
    console.log("进行图片展示", imgs);
  };

  const onHandleClick = (info: any) => {
    console.log("进行跳转", info);
  };

  return (
    <Comment
      images={cmt.images}
      videos={cmt.videos}
      info={cmt.info}
      onClick={onHandleClick}
      onClickImages={onClickImages}
      onClickOperate={(type: string) => {
        console.log(type);
      }}
      operation={["reply"]}
      commentLabels={
        <img
          style={{
            height: "20px",
          }}
          src="https://img11.360buyimg.com/imagetools/jfs/t1/211858/17/4258/12101/618e6f78Ed0edcadc/e83a673555edf59f.jpg"
        />
      }
    ></Comment>
  );
};
export default App;
```
:::

### Multi Line Image

The value of `type` allows you to set the image to be displayed in multiple lines

:::demo 
```js
import React from 'react';
import { Comment } from '@nutui/nutui-biz';

const App = () => {

  const [cmt, setCmt] = useState({} as any);

  const getData = () => {
    fetch("//storage.360buyimg.com/nutui/3x/comment_data.json")
      .then((response) => response.json())
      .then((res) => {
        res.Comment.info.avatar =
          "https://img14.360buyimg.com/imagetools/jfs/t1/167902/2/8762/791358/603742d7E9b4275e3/e09d8f9a8bf4c0ef.png";
        setCmt(res.Comment);
      })
      .catch((err) => console.log("Oh, error", err));
  };

  useEffect(() => {
    getData();
  }, []);

  const onClickImages = (imgs: any) => {
    console.log("进行图片展示", imgs);
  };

  const onHandleClick = (info: any) => {
    console.log("进行跳转", info);
  };

  return (
    <Comment
      type="complex"
      imagesRows="multi"
      images={cmt.images}
      videos={cmt.videos}
      info={cmt.info}
      ellipsis="6"
      onClickImages={onClickImages}
      commentLabels={
        <img
          src="https://storage.360buyimg.com/imgtools/78925d9440-f9e874d0-e93d-11eb-8e5c-0da9e18a13b1.png"
          style={{ height: "12px" }}
        />
      }
      commentShopReply={
        <div className="nut-comment-shop">
          <span>京东美妆国际：</span>
          尊敬的客户您好，非常抱歉给您带来不愉快的购物体验，关于过敏，什么成分都不存在个别性和普遍性。
        </div>
      }
    ></Comment>
  );
};
export default App;
```
:::

### Additional Review

:::demo 
```js
import React from 'react';
import { Comment } from '@nutui/nutui-biz';

const App = () => {

  const [cmt, setCmt] = useState({} as any);

  const getData = () => {
    fetch("//storage.360buyimg.com/nutui/3x/comment_data.json")
      .then((response) => response.json())
      .then((res) => {
        res.Comment.info.avatar =
          "https://img14.360buyimg.com/imagetools/jfs/t1/167902/2/8762/791358/603742d7E9b4275e3/e09d8f9a8bf4c0ef.png";
        setCmt(res.Comment);
      })
      .catch((err) => console.log("Oh, error", err));
  };

  useEffect(() => {
    getData();
  }, []);

  const onClickImages = (imgs: any) => {
    console.log("进行图片展示", imgs);
  };

  return (
    <Comment
      imagesRows="multi"
      images={cmt.images}
      videos={cmt.videos}
      info={cmt.info}
      follow={cmt.follow}
      onClickImages={onClickImages}
    ></Comment>
  );
};
export default App;
```
:::
## API

### Props

| Attribute            | Description               | Type   | Default  |
|--------------|----------------------------------|--------|------------------|
| type         | Header type      | string | `default`               |
| imagesRows         | Comment picture shows type of lines    | string | `one`               |
| ellipsis        | Ellipsis number       | string \| number | `2`               |
| videos         | Videos data | Array | `[]`              |
| images | Images data    | Array | `[]` |
| info     | Comment data                      | object | `{}`             |
| follow     | Additional review data                      | object | `{}`             |
| operation | Bottom button   | Array | `["reply", "like", "more"]` |
| comment-labels  | Comment user label | ReactNode | - |
| comment-shop-reply  | At the bottom of the comment is the display of the merchant's response | ReactNode | - |


### Events

| Event            | Description               | Arguments   |
|--------|----------------|--------------|
| onClickOperate  | Emitted when to click bottom button | `type`, like `operation`  |
| onClick  | Emitted when to click comment | The `info` parameter is passed in |
| onClickImages | Emitted when to click images or videos | `{type,index,value}` |

### images data

```javascript
const images = [
  {
    id: "", // key
    smallImgUrl: "", // Small picture, used for table presentation
    bigImgUrl: "", // Big picture, big picture display use
    imgUrl: "", // Out figure
  }
]
```

### videos data

```javascript
const videos = [
  {
    id: "", // key
    mainUrl: "", // Video mask image
    videoUrl: "", // Video url
  }
]
```

### info data

```javascript
const info = {
  content: "",  // Comment on the details
  nickName: "", // The name of the reviewer
  score: 5, // Number of comment stars
  avatar: "",  // The avatar of the reviewer
  time: "", // Comment time
  size: "", // Comment good size
  reply:23, // Number of replies to this comment
  like:1, // Number of likes for this comment
}
```

### follow data

Information related to review.

```javascript
const follow = {
  days: 0, // Review how many days after purchase
  content: "", // Review content
  images: [] // Review images
}
```
