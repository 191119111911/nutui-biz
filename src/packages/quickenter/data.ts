export const quickEnterData = [
  {
    displayName: "五折专区",
    imageUrl:
      "https://img10.360buyimg.com/imagetools/s100x100_jfs/t1/171408/20/5251/89724/601919feE621f66a5/79fd92576b72cfd4.gif",
  },
  {
    displayName: "手机通讯",
    imageUrl:
      "https://img12.360buyimg.com/imagetools/s100x100_jfs/t1/135346/37/5276/9667/5f1a50a3E51cf1d21/2831b07712127aaa.png",
  },
  {
    displayName: "米面粮油",
    imageUrl:
      "https://img30.360buyimg.com/imagetools/s100x100_jfs/t1/112915/31/1239/9586/5e9586eaE549f1ecb/75ee4f10fdb43a1d.png",
  },
  {
    displayName: "时令生鲜",
    imageUrl:
      "https://img12.360buyimg.com/imagetools/s100x100_jfs/t1/128638/35/7916/14107/5f1a50a3E316b5020/e25dd3e7d15c835d.png",
  },
  {
    displayName: "酒水饮料",
    imageUrl:
      "https://img11.360buyimg.com/imagetools/s100x100_jfs/t1/126267/19/7965/10474/5f1a50a3E9dbf8c1e/1ccdc3b1f6f14fe7.png",
  },
  {
    displayName: "烹饪厨具",
    imageUrl:
      "https://img20.360buyimg.com/imagetools/s100x100_jfs/t1/136473/9/5269/10071/5f1a50a3E211837ab/88345f88c8242cd3.png",
  },
  {
    displayName: "美妆护肤",
    imageUrl:
      "https://img10.360buyimg.com/imagetools/s100x100_jfs/t1/142203/8/3785/10276/5f1a50a3E7b0efbc5/b156c58c52dc734d.png",
  },
  {
    displayName: "清洁纸品",
    imageUrl:
      "https://img11.360buyimg.com/imagetools/s100x100_jfs/t1/138185/13/3701/9381/5f1a50a3Ea36e2578/ad39b6bd9007c009.png",
  },
  {
    displayName: "床上用品",
    imageUrl:
      "https://img11.360buyimg.com/imagetools/s100x100_jfs/t1/122764/9/7920/11265/5f1a50a3Ea5580ddb/7dd7b2b5fc497e4b.png",
  },
  {
    displayName: "钟表奢品",
    imageUrl:
      "https://img13.360buyimg.com/imagetools/s100x100_jfs/t1/122338/17/8034/12430/5f1a50a3Ec3d71123/84f59b915f232b5b.png",
  },

  {
    displayName: "生日祝福",
    imageUrl:
      "https://m.360buyimg.com/da/s88x88_jfs/t1/81435/32/20228/6445/62bd2e5eE75b5e516/e30deb58d149febe.png.webp",
  },
  {
    displayName: "国庆节",
    imageUrl:
      "https://m.360buyimg.com/da/s88x88_jfs/t1/39449/24/16764/6031/62bd2e5eEa016808c/9a34b161b82f4483.png.webp",
  },
  {
    displayName: "端午节",
    imageUrl:
      "https://m.360buyimg.com/da/s88x88_jfs/t1/205959/11/22745/8368/62bd2e5eEdaeed436/f04fa37ba70361f8.png.webp",
  },
  {
    displayName: "健康防疫",
    imageUrl:
      "https://m.360buyimg.com/da/s88x88_jfs/t1/21220/3/17668/4806/62bda22cE0068d2b9/bdc93be39ff2a76b.png.webp",
  },
  {
    displayName: "食堂饭卡",
    imageUrl:
      "https://m.360buyimg.com/da/s88x88_jfs/t1/199375/3/25838/6556/62bda088Ee9b0896c/91619e1e5f96fc40.png.webp",
  },
];
