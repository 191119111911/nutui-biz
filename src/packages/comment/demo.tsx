import React, { useState, useEffect } from "react";
import { Comment } from "./comment";
import { Cell, Tag } from "@nutui/nutui-react";
import { useTranslate } from "../../sites/assets/locale";

interface T {
  [props: string]: string;
}

const CommentDemo = () => {
  const [translated] = useTranslate<T>({
    "zh-CN": {
      basic: "基础用法",
      single: "评论图片单行展示",
      multiRow: "评论图片多行展示",
      additionalReviewd: "追评展示",
    },
    "en-US": {
      basic: "Basic Usage",
      single: "Single Line Image",
      multiRow: "Multi Line Image ",
      review: "Additional Review",
    },
  });

  const [cmt, setCmt] = useState({} as any);

  const getData = () => {
    fetch("//storage.360buyimg.com/nutui/3x/comment_data.json")
      .then((response) => response.json())
      .then((res) => {
        res.Comment.info.avatar =
          "https://img14.360buyimg.com/imagetools/jfs/t1/167902/2/8762/791358/603742d7E9b4275e3/e09d8f9a8bf4c0ef.png";
        setCmt(res.Comment);
      })
      .catch((err) => console.log("Oh, error", err));
  };

  useEffect(() => {
    getData();
  }, []);

  const onClickImages = (imgs: any) => {
    console.log("进行图片展示", imgs);
  };

  const onHandleClick = (info: any) => {
    console.log("进行跳转", info);
  };

  return (
    <>
      <div className="demo">
        <h2>{translated.single}</h2>
        <Cell>
          <Comment
            images={cmt.images}
            videos={cmt.videos}
            info={cmt.info}
            onClick={onHandleClick}
            onClickImages={onClickImages}
            onClickOperate={(type: string) => {
              console.log(type);
            }}
            operation={["reply"]}
            commentLabels={
              <img
                style={{
                  height: "20px",
                }}
                src="https://img11.360buyimg.com/imagetools/jfs/t1/211858/17/4258/12101/618e6f78Ed0edcadc/e83a673555edf59f.jpg"
              />
            }
          ></Comment>
        </Cell>

        <h2>{translated.multiRow}</h2>
        <Cell>
          <Comment
            type="complex"
            imagesRows="multi"
            images={cmt.images}
            videos={cmt.videos}
            info={cmt.info}
            ellipsis="6"
            onClickImages={onClickImages}
            commentLabels={
              <img
                src="https://storage.360buyimg.com/imgtools/78925d9440-f9e874d0-e93d-11eb-8e5c-0da9e18a13b1.png"
                style={{ height: "12px" }}
              />
            }
            commentShopReply={
              <div className="nut-comment-shop">
                <span>京东美妆国际：</span>
                尊敬的客户您好，非常抱歉给您带来不愉快的购物体验，关于过敏，什么成分都不存在个别性和普遍性。
              </div>
            }
          ></Comment>
        </Cell>

        <h2>{translated.additionalReviewd}</h2>
        <Cell>
          <Comment
            imagesRows="multi"
            images={cmt.images}
            videos={cmt.videos}
            info={cmt.info}
            follow={cmt.follow}
            onClickImages={onClickImages}
          ></Comment>
        </Cell>
      </div>
    </>
  );
};

export default CommentDemo;
