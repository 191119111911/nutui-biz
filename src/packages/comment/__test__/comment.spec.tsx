import * as React from 'react'

import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Comment } from '../comment'
import data from '../data.json'

test('props & one imageRow test', () => {
  const state = data.Comment
  state.info.avatar = "https://img14.360buyimg.com/imagetools/jfs/t1/167902/2/8762/791358/603742d7E9b4275e3/e09d8f9a8bf4c0ef.png"
  const onHandleClick = (info: any) => {
    console.log("进行跳转", info)
  }
  const onClickImages = (imgs: any) => {
    console.log("进行图片展示", imgs)
  }
  const { container } = render(
    <Comment
      images={state.images}
      videos={state.videos}
      info={state.info}
      onClick={onHandleClick}
      onClickImages={onClickImages}
      onClickOperate={(type: string) => {
        console.log(type);
      }}
      operation={["reply"]}
      commentLabels={
        <img
          className='item'
          style={{
            height: '20px'
          }}
          src="https://img11.360buyimg.com/imagetools/jfs/t1/211858/17/4258/12101/618e6f78Ed0edcadc/e83a673555edf59f.jpg"
        />
      }
    ></Comment>
  )
  // rate
  const nutRate = container.querySelector('.nut-rate') as HTMLElement
  expect(nutRate).toBeTruthy()
  // avatar
  expect(
    container.querySelector('.nut-comment-header__user-avter img')?.getAttribute('src')
  ).toBe(state.info.avatar)
  // nickName
  expect(container.querySelector('.nut-comment-header__user-default-name')).toContainHTML(
    state.info.nickName
  )
  // time
  expect(container.querySelector('.nut-comment-header__time')).toContainHTML(
    state.info.time
  )
  // videos
  expect(container.querySelectorAll('.nut-comment-images__item--video').length).toBe(state.videos.length)
  // images
  expect(container.querySelectorAll('.nut-comment-images__item--imgbox').length).toBe(state.images.length)
  // commentLabels
  expect(container.querySelector('.nut-comment-header__user-default img.item')).toHaveStyle({
    height: '20px'
  })
  // one
  const images = container.querySelector('.nut-comment-images--one') as HTMLElement
  expect(images).toBeTruthy()
  expect(container).toMatchSnapshot()
})

test('multi imagesRows test', () => {
  const state = data.Comment
  state.info.avatar = "https://img14.360buyimg.com/imagetools/jfs/t1/167902/2/8762/791358/603742d7E9b4275e3/e09d8f9a8bf4c0ef.png"
  const onClickImages = jest.fn()
  const onClickOperate = jest.fn()
  const { container } = render(
    <Comment
      type="complex"
      imagesRows="multi"
      images={state.images}
      videos={state.videos}
      info={state.info}
      ellipsis="6"
      onClickOperate={onClickOperate}
      onClickImages={onClickImages}
      commentLabels={
        <img
          src="https://storage.360buyimg.com/imgtools/78925d9440-f9e874d0-e93d-11eb-8e5c-0da9e18a13b1.png"
          style={{ height: "12px" }}
        />
      }
      commentShopReply={
        <div className="nut-comment-shop">
          <span>京东美妆国际：</span>
          尊敬的客户您好，非常抱歉给您带来不愉快的购物体验，关于过敏，什么成分都不存在个别性和普遍性。
        </div>
      }
    ></Comment>
  )
  // multi images
  const images = container.querySelector('.nut-comment-images--multi') as HTMLElement
  expect(images).toBeTruthy()
  // click video
  const video = container.querySelector('.nut-comment-images__item--video') as HTMLElement
  fireEvent.click(video)
  expect(onClickImages).toBeCalledWith({
    index: 0,
    type: 'video',
    value: state.videos[0]
  })
  // operations
  const operations = container.querySelectorAll('.nut-comment-bottom__cpx-item')
  expect(operations.length).toBe(3)
  // click operate reply
  const reply = container.querySelector('.nut-comment-bottom__cpx-item--reply') as HTMLElement
  fireEvent.click(reply)
  expect(onClickOperate).toBeCalledWith('reply')
  // click operate more
  const more = container.querySelector('.nut-comment-bottom__cpx-item--more') as HTMLElement
  fireEvent.click(more)
  expect(onClickOperate).toBeCalledWith('more')
  expect(container).toMatchSnapshot()
})

test('follow test', () => {
  const state = data.Comment
  state.info.avatar = "https://img14.360buyimg.com/imagetools/jfs/t1/167902/2/8762/791358/603742d7E9b4275e3/e09d8f9a8bf4c0ef.png"
  const onClick = jest.fn()
  const { container } = render(
    <Comment
      type='complex'
      imagesRows="multi"
      images={state.images}
      videos={state.videos}
      info={state.info}
      follow={state.follow}
      onClick={onClick}
    ></Comment>
  )
  // follow
  const follow = container.querySelector('.nut-comment__follow') as HTMLElement
  expect(follow).toBeTruthy
  // click reply
  const replyMain = container.querySelector('.nut-comment__main') as HTMLElement
  fireEvent.click(replyMain)
  expect(onClick).toBeCalled()
})