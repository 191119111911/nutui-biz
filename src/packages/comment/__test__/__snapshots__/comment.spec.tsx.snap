// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multi imagesRows test 1`] = `
<div>
  <div
    class="nut-comment"
  >
    <div>
      <div
        class="nut-comment-header"
      >
        <div
          class="nut-comment-header__user"
        >
          <div
            class="nut-comment-header__user-avter"
          >
            <img
              src="https://img14.360buyimg.com/imagetools/jfs/t1/167902/2/8762/791358/603742d7E9b4275e3/e09d8f9a8bf4c0ef.png"
            />
          </div>
          <div
            class="nut-comment-header__user-complex"
          >
            <span
              class="nut-comment-header__user-complex-name"
            >
              星***_
            </span>
            <img
              src="https://storage.360buyimg.com/imgtools/78925d9440-f9e874d0-e93d-11eb-8e5c-0da9e18a13b1.png"
              style="height: 12px;"
            />
          </div>
        </div>
        <div
          class="nut-comment-header__time"
        >
          2021-09-09
        </div>
      </div>
      <div
        class="nut-comment-header__complex-score"
      >
        <div
          class="nut-rate undefined"
        >
          <div
            class="nut-rate-item"
            style="margin-right: 3px;"
          >
            <i
              class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
              style="font-size: 12px; height: 12px; width: 12px;"
            />
          </div>
          <div
            class="nut-rate-item"
            style="margin-right: 3px;"
          >
            <i
              class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
              style="font-size: 12px; height: 12px; width: 12px;"
            />
          </div>
          <div
            class="nut-rate-item"
            style="margin-right: 3px;"
          >
            <i
              class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
              style="font-size: 12px; height: 12px; width: 12px;"
            />
          </div>
          <div
            class="nut-rate-item"
            style="margin-right: 3px;"
          >
            <i
              class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
              style="font-size: 12px; height: 12px; width: 12px;"
            />
          </div>
          <div
            class="nut-rate-item"
            style="margin-right: 3px;"
          >
            <i
              class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
              style="font-size: 12px; height: 12px; width: 12px;"
            />
          </div>
        </div>
        <i
          class="nut-comment-header__complex-score-i"
        />
        <div
          class="nut-comment-header__complex-score-size"
        >
          雪域白【碎屏险套装】全网通；8G+256【12期免息】
        </div>
      </div>
    </div>
    <div
      class="nut-comment__main"
    >
       
      产品包装：塑料瓶，但是能接受，也很喜欢。
       
    </div>
    <div
      class="nut-comment-images nut-comment-images--multi"
    >
      <div
        class="nut-comment-images__item nut-comment-images__item--video"
      >
        <img
          src="https://img.300hu.com/4c1f7a6atransbjngwcloud1oss/3a5abb98377496264567160833/imageSampleSnapshot/1604738160_777980391.100_1344.jpg"
        />
        <div
          class="nut-comment-images__play"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/153847/3/5051/62777/5fa65e01Ec88cecb9/36ba206949050b39.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/131794/19/15190/57841/5fa65e03E75c12ef1/0d1ac3a87ff9e398.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/155091/12/4926/68141/5fa65e03E9d8c5120/9c71b6e49f1eb3c3.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/138774/29/13635/70364/5fa65e04E1669ad70/c97839098a18d92e.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/120265/1/17811/70109/5fa65e05Eb23a84e3/e156fc1d316922c4.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/120265/1/17811/70109/5fa65e05Eb23a84e3/e156fc1d316922c4.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/155091/12/4926/68141/5fa65e03E9d8c5120/9c71b6e49f1eb3c3.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/138774/29/13635/70364/5fa65e04E1669ad70/c97839098a18d92e.jpg"
        />
        <div
          class="nut-comment-images__mask"
        >
          <span>
            共 
            10
             张
          </span>
          <i
            class="nutui-iconfont nut-icon nut-icon-right "
            style="font-size: 12px; height: 12px; width: 12px;"
          />
        </div>
      </div>
    </div>
    <div
      class="nut-comment-bottom"
    >
      <div
        class="nut-comment-bottom__lable"
      />
      <div
        class="nut-comment-bottom__cpx"
      >
        <div
          class="nut-comment-bottom__cpx-item nut-comment-bottom__cpx-item--reply"
        >
          <span>
            23
          </span>
          <i
            class="nutui-iconfont nut-icon nut-icon-comment "
          />
        </div>
        <div
          class="nut-comment-bottom__cpx-item nut-comment-bottom__cpx-item--like"
        >
          <span>
            1
          </span>
          <i
            class="nutui-iconfont nut-icon nut-icon-fabulous "
          />
        </div>
        <div
          class="nut-comment-bottom__cpx-item nut-comment-bottom__cpx-item--more"
        >
          <i
            class="nutui-iconfont nut-icon nut-icon-more-x "
          />
          <div
            class="nut-comment-bottom__cpx-item-popover"
          >
            我要投诉
          </div>
        </div>
      </div>
    </div>
    <div
      class="nut-comment-shop"
    >
      <span>
        京东美妆国际：
      </span>
      尊敬的客户您好，非常抱歉给您带来不愉快的购物体验，关于过敏，什么成分都不存在个别性和普遍性。
    </div>
  </div>
</div>
`;

exports[`props & one imageRow test 1`] = `
<div>
  <div
    class="nut-comment"
  >
    <div>
      <div
        class="nut-comment-header"
      >
        <div
          class="nut-comment-header__user"
        >
          <div
            class="nut-comment-header__user-avter"
          >
            <img
              src="https://img14.360buyimg.com/imagetools/jfs/t1/167902/2/8762/791358/603742d7E9b4275e3/e09d8f9a8bf4c0ef.png"
            />
          </div>
          <div
            class="nut-comment-header__user-default"
          >
            <div
              class="nut-comment-header__user-default-name"
            >
              <span>
                星***_
              </span>
              <img
                class="item"
                src="https://img11.360buyimg.com/imagetools/jfs/t1/211858/17/4258/12101/618e6f78Ed0edcadc/e83a673555edf59f.jpg"
                style="height: 20px;"
              />
            </div>
            <div
              class="nut-comment-header__user-score"
            >
              <div
                class="nut-rate undefined"
              >
                <div
                  class="nut-rate-item"
                  style="margin-right: 5px;"
                >
                  <i
                    class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
                    style="font-size: 12px; height: 12px; width: 12px;"
                  />
                </div>
                <div
                  class="nut-rate-item"
                  style="margin-right: 5px;"
                >
                  <i
                    class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
                    style="font-size: 12px; height: 12px; width: 12px;"
                  />
                </div>
                <div
                  class="nut-rate-item"
                  style="margin-right: 5px;"
                >
                  <i
                    class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
                    style="font-size: 12px; height: 12px; width: 12px;"
                  />
                </div>
                <div
                  class="nut-rate-item"
                  style="margin-right: 5px;"
                >
                  <i
                    class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
                    style="font-size: 12px; height: 12px; width: 12px;"
                  />
                </div>
                <div
                  class="nut-rate-item"
                  style="margin-right: 5px;"
                >
                  <i
                    class="nutui-iconfont nut-icon nut-icon-star-fill-n nut-rate-item__icon "
                    style="font-size: 12px; height: 12px; width: 12px;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="nut-comment-header__time"
        >
          2021-09-09
        </div>
      </div>
    </div>
    <div
      class="nut-comment__main"
    >
       
      产品包装：塑料瓶，但是能接受，也很喜欢。
       
    </div>
    <div
      class="nut-comment-images nut-comment-images--one"
    >
      <div
        class="nut-comment-images__item nut-comment-images__item--video"
      >
        <img
          src="https://img.300hu.com/4c1f7a6atransbjngwcloud1oss/3a5abb98377496264567160833/imageSampleSnapshot/1604738160_777980391.100_1344.jpg"
        />
        <div
          class="nut-comment-images__play"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/153847/3/5051/62777/5fa65e01Ec88cecb9/36ba206949050b39.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/131794/19/15190/57841/5fa65e03E75c12ef1/0d1ac3a87ff9e398.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/155091/12/4926/68141/5fa65e03E9d8c5120/9c71b6e49f1eb3c3.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/138774/29/13635/70364/5fa65e04E1669ad70/c97839098a18d92e.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/120265/1/17811/70109/5fa65e05Eb23a84e3/e156fc1d316922c4.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/120265/1/17811/70109/5fa65e05Eb23a84e3/e156fc1d316922c4.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/155091/12/4926/68141/5fa65e03E9d8c5120/9c71b6e49f1eb3c3.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/138774/29/13635/70364/5fa65e04E1669ad70/c97839098a18d92e.jpg"
        />
      </div>
      <div
        class="nut-comment-images__item nut-comment-images__item--imgbox"
      >
        <img
          src="https://img30.360buyimg.com/imagetools/jfs/t1/138774/29/13635/70364/5fa65e04E1669ad70/c97839098a18d92e.jpg"
        />
      </div>
    </div>
    <div
      class="nut-comment-bottom"
    >
      <div
        class="nut-comment-bottom__lable"
      >
        <span>
          雪域白【碎屏险套装】全网通；8G+256【12期免息】
        </span>
      </div>
      <div
        class="nut-comment-bottom__cpx"
      >
        <div
          class="nut-comment-bottom__cpx-item nut-comment-bottom__cpx-item--reply"
        >
          <span>
            23
          </span>
          <i
            class="nutui-iconfont nut-icon nut-icon-comment "
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
